pkg/www/retext/src/core/Channel.js

```diff
@@ -0,0 +1,33 @@
+class Packet {
+    constructor(data, callback) {
+        this.data = data;
+        this.callback = callback;
+    }
+}
+
+export class Channel {
+    constructor() {
+        this._packet = false;
+    }
+
+    /**
+     * Must be called receive or the data will be ignored/overwritten on the next call to send()
+     *
+     * @param data
+     * @param sendCallback
+     */
+    send(data, sendCallback) {
+        this._packet = new Packet(data, sendCallback);
+    }
+
+    /**
+     * Must be called after send or it'll just return false. Clears out the package after the first use.
+     *
+     * @returns {boolean|object}
+     */
+    receive() {
```

Comment:
What happens if receive is called before send? _packet should probably have an empty flag that is set to True in the case that this._packet isnt' set. I also think that null is a slightly better indicator of something not being set than false. Since false sorta implies that this._packet is two different types (bool or Packet), as opposed to being (set Packet or not set Null).